File: _archive/lararium_mcp/diagnostics.py
"""Diagnostics for the Lararium MCP carrier spine."""

from __future__ import annotations

from dataclasses import dataclass
from typing import Literal

Severity = Literal["info", "warning", "error"]


@dataclass(frozen=True)
class Diagnostic:
    """A small, serializable compiler diagnostic."""

    code: str
    message: str
    severity: Severity = "error"

    def to_dict(self) -> dict[str, str]:
        return {
            "code": self.code,
            "message": self.message,
            "severity": self.severity,
        }

File: packages/lararium-core/maybe/diagnostics.ts
export type Severity = "info" | "warning" | "error";

export interface Diagnostic {
  readonly code: string;
  readonly message: string;
  readonly severity: Severity;
}

export function makeDiagnostic(code: string, message: string, severity: Severity = "error"): Diagnostic {
  return { code, message, severity };
}
